builtin/mainmenu/tab_local.lua

```diff
@@ -242,6 +246,14 @@ local function main_button_handler(this, fields, name, tabdata)
 		return true
 	end
 
+	if fields["te_playername"] then
+		current_name = fields["te_playername"]
+	end
+
+	if fields["te_serverport"] then
+		current_port = fields["te_serverport"]
+	end
+
```

Comment:
Switching the selected world still resets the boxes. You should put this code before `if fields["sp_worlds"] ~= nil then`